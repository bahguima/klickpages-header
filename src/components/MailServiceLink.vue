<template>
<div class="nav-mail">
  <a :href="topBarConfig.mailservice.url" class="lock-screen">
    <mail-service-icon />
  </a>
  <div class="tooltip-status">
    <div class="tooltip-inner">
      {{topBarConfig.mailservice.description}}
    </div>
  </div>
</div>
</template>

<script>
import { mapGetters } from 'vuex';
import MailServiceIcon from './icons/MailService.vue';

export default {
  computed: mapGetters({
    topBarConfig: 'topBar/config',
  }),
  components: {
    MailServiceIcon,
  },
};
</script>
