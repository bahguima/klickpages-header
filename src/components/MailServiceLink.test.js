import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import MailServiceLink from './MailServiceLink.vue';

describe('components/MailServiceLink.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  const topBarConfig = {
    mailservice: {
      url: 'mailservice-url.com',
      description: 'mailservice description',
    },
  };

  const topBar = {
    namespaced: true,
    getters: {
      config: () => topBarConfig,
    },
  };

  let wrapper;

  describe('when component is mounted', () => {
    beforeAll(() => {
      const store = new Vuex.Store({ modules: { topBar } });
      wrapper = shallowMount(MailServiceLink, { localVue, store });
    });

    afterAll(() => {
      wrapper.destroy();
      jest.clearAllMocks();
    });

    it('should render the html correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
