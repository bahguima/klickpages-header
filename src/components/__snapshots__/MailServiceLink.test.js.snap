// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MailServiceLink.vue when component is mounted should render the html correctly 1`] = `
<div
  class="nav-mail"
>
  <a
    class="lock-screen"
    href="mailservice-url.com"
  >
    <mail-service-icon-stub />
  </a>
   
  <div
    class="tooltip-status"
  >
    <div
      class="tooltip-inner"
    >
      
      mailservice description
    
    </div>
  </div>
</div>
`;
